RecoVertex/BeamSpotProducer/src/BeamFitter.cc

```diff
@@ -684,7 +684,7 @@ void BeamFitter::dumpTxtFile(std::string & fileName, bool append){
     currentBS.endLumiOfFit   = fendLumiOfFit  ;
     std::copy(freftime, freftime+2, currentBS.reftime);
           
-    beamspot::dumpBeamSpotTxt(fileName, append, currentBS);
+    if (outFile.is_open()) beamspot::dumpBeamSpotTxt(outFile, currentBS);
```

Comment:
what is the purpose of this: how can a file not be open in the current context given that it's the first access to the file after the above calls
```C++
   if (!append)
     outFile.open(fileName.c_str());
   else
     outFile.open(fileName.c_str(),std::ios::app);
```

was the bug due to concurrent writing to the same file from multiple threads? if so, then this doesn't seem like the right thing to fix the problem.